pkg/metrics/metric/metric.go

```diff
@@ -189,3 +189,10 @@ type Opts struct {
 	// If true, the metric has to be explicitly enabled via config or flags
 	Disabled bool
 }
+
+func (b Opts) GetConfigName() string {
+	if b.ConfigName == "" {
+		return prometheus.BuildFQName(b.Namespace, b.Subsystem, b.Name)
```

Comment:
I wonder if we should enforce that these are never empty

Reply:
So the only hard requirement is that `Name` is non-empty. That's why I used `BuildFQName` instead of regular `strings.Join`